### ISSUE TITLE
Add `syntax-string-no-invalid`

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#8213]
### What is the problem you're trying to solve?

There are specific rules for [syntax strings](https://drafts.css-houdini.org/css-properties-values-api/#syntax-strings).

```css
@property --foo {
  syntax: "this-string"
}
```

It's easy for people to make mistakes when defining them.

Ref: https://github.com/stylelint/stylelint/pull/8211#discussion_r1892711278

### What solution would you like to see?

A new rule to flag invalid syntax strings.

I don't even know if validating the syntax is viable, but I figured it was worth discussing.

It could start simple and only flag [unsupported names](https://drafts.css-houdini.org/css-properties-values-api/#supported-names), for example:

```css
@property --foo {
  syntax: "<alpha-value>"
}
```

**Comment by jeddy3** [comment]
This may be viable using [CSSTree's syntax parser](https://github.com/csstree/csstree/blob/master/docs/definition-syntax.md#parsesource), including [walking](https://github.com/csstree/csstree/blob/master/docs/definition-syntax.md#walknode-options-context) to find unsupported names.

**Comment by jeddy3** [comment]
Seems suitable:

```js
import { definitionSyntax } from "css-tree";

const supportedTypes = new Set(["color", "length"]);

let ast;
try {
  ast = definitionSyntax.parse("<alpha-value>");
  definitionSyntax.walk(ast, (node) => {
    const { name, type } = node;
    if (type === "Type" && !supportedTypes.has(name)) {
      console.error(`Unsupported type name: ${name}`);
    }
  });
} catch (error) {
  console.error("Parse error");
}
```

It'll catch `<alpha-value>` and parse errors like unbalanced brackets and malformed multipliers.

**Comment by ybiquitous** [comment]
Sounds good! And it seems we can implement this idea, thanks to the existing CSSTree syntax parser. 👍🏼

**Comment by jeddy3** [comment]
- Name: `syntax-string-no-invalid`
- Description: Disallow invalid syntax strings.
- Message: `Unexpected invalid syntax string "${syntaxString}"`
- Primary option: `true`
- Autofix: No
- Category: Avoid errors > Invalid
- Included in sharable configs?: Yes
- Expanded description: Syntax strings are used for the `syntax` descriptor value of the `@property` at-rule. This rule checks their grammar and flags unsupported type names.
